editor/src/messages/portfolio/document/node_graph/node_graph_message_handler/document_node_types.rs

```diff
@@ -1667,6 +1667,14 @@ fn static_nodes() -> Vec<DocumentNodeType> {
 			outputs: vec![DocumentOutputType::new("Image", FrontendGraphDataType::Raster)],
 			properties: node_properties::index_node_properties,
 		},
+		DocumentNodeType {
+			name: "Image Color Palette",
+			category: "Vector",
+			identifier: NodeImplementation::proto("graphene_core::raster::ImageColorPaletteNode"),
+			inputs: vec![DocumentInputType::value("Image", TaggedValue::ImageFrame(ImageFrame::empty()), true)],
+			outputs: vec![DocumentOutputType::new("Vector", FrontendGraphDataType::Color)],
```

Comment:
I didn't understand totally how the system works, but I have a particular issue about the `outputs`. How it works, how this output is used in the editor and how I see it on there. Furthermore, how should I use the `FrontendGraphDataType::Color`? I have looked in the code looking for examples, but I found none using it.

Reply:
The outputs here is purely for the UI.

The `FrontendGraphDataType` determines the colour of the dot that represents the output in the UI. For example vector data is blue, and numbers are pink. This will in future be determined by the type system, however we haven't got around to re-factoring that yet, so for now it is hardcoded for each node type. 

The output's name is shown next to the dot in the UI, however it is not shown for the first output. The value for the first output is currently not used at all; although it is still probably a good idea to change it to "Palette" or similar in case the UI is changed in future.

![image](https://github.com/GraphiteEditor/Graphite/assets/78500760/8ef122f2-1fbb-414f-962b-161930c3f9cd)